File: server/Models/Recipe.cs
namespace AllSpice.Models;

public class Recipe : RepoItem<int>
{
    public string Title { get; set; }

    public string Instructions { get; set; }

    public string Img { get; set; }

    public string Category { get; set; }

    public string CreatorId { get; set; }

    public Account Creator { get; set; }

}
public class FavoriteRecipe : Recipe
{
    public int FavoriteId { get; set; }
}

File: client/src/models/Recipe.js
export class Recipe {
    constructor(data) {
        this.img = data.img
        this.title = data.title
        this.instructions = data.instructions
        this.category = data.category
        this.creatorId = data.creatorId
        this.creator = data.creator
        this.id = data.id
    }
}

export class RecipeIngredient extends Recipe {
    constructor(data) {
        super(data)
        this.name = data.name
        this.quantity = data.quantity
    }
}